src/pybind/mgr/dashboard/controllers/daemon.py

```diff
@@ -31,3 +31,11 @@ def set(self, daemon_name: str, action: str = '',
         orch = OrchClient.instance()
         res = orch.daemons.action(action=action, daemon_name=daemon_name, image=container_image)
         return res
+
+    @raise_if_no_orchestrator([OrchFeature.DAEMON_LIST])
+    @handle_orchestrator_error('daemon')
+    @RESTController.MethodMap(version=APIVersion.DEFAULT)
+    def list(self):
+        orch = OrchClient.instance()
+        daemons = orch.services.list_daemons()
+        return [d.to_dict() for d in daemons]
```

Comment:
How's scalable is this endpoint? On a very-large cluster (Pawsey-like) this may be 8k items.

Reply:
We've not got issues on the daemon endpoint so far. We get the cached result from cephadm. And no processing as well! But I think this is the first time we are listing the entire daemons in a cluster (usually we filter them out either via hostname or service_name). So will need to test this with huge numbers to get some answers! 
@pereman2 do you have any thoughts here?